cypress/support/verifyFormValue.js

```diff
@@ -0,0 +1,7 @@
+Cypress.Commands.add('verifyFormValue', (fieldName, fieldValue) => {
+    // make sure form spy function ran
+    cy.get('.form-spy-internal')
+    cy.window().then(win => {
+        expect(win.formValues[fieldName]).to.equal(fieldValue)
```

Comment:
This is going to fail once we have more complex form values (arrays, files, objects through format/parse).

Maybe we can do this slightly differently:


```js
const { Promise } = Cypress

const getFormValue = fieldName => {
  cy.get('.form-spy-internal')
  return cy.window().then(win => {
    if (win.formValues.hasOwnProperty(fieldName) {
      return Promise.reject(`"${fieldName} not present on "window.formValues"`)
    }
    
    return Promise.resolve(win.formValues[fieldName])
  }
}

Cypress.Commands.add('getFormValue', getFormValue)
```

Then we can do the following:

```js
cy.getFormValue('tnc').then(tnc => expect(tnc).to.be.undefined)
```

From there we could still create other helpers, although I prefer the above example.

```js
const verifyFormValueEquals = (fieldName, expected) => {
  cy.getFormValue(fieldName).then(actual =>
    expect(actual).to.equal(expected)
  )
}
```

Reply:
EDIT: See #146, I've added it there

Reply:
> This is going to fail once we have more complex form values (arrays, files, objects through format/parse).

I hadn't considered `format/parse` scenarios with objects. You are right these would fail.

But just to confirm: once #146 is merged in, this discussion is no longer relevant right?

Reply:
> once #146 is merged in, this discussion is no longer relevant right?

Yes

> I hadn't considered format/parse scenarios with objects. You are right these would fail.

Even without, it'd fail with selects and *Groups

Reply:
#146 was merged into this branch, so I am resolving this conversation